### DISCUSSION TITLE
Creating a bundle, that exports Class-Structure into JSON format

### DISCUSSION THREAD

**Discussion by DenisKudic** [pimcore/pimcore#13686] [Q&A / Support]
Hello everyone,

i'm a fresh junior developer in the pimcore world and i already successfully completed the junior pimcore-academy final exam.
Now i would like to apply my knowledge on my own and for my first task, i want to export the class structure in a defined .json file, which the admin user can download.

For that, i want to create a simple UI in the admin interface, where the user only has to click one button to create a process, where the backend generates a .json format of the given data. 

Unfortunatelly i dont know where i have to start exactly. I already found a bundle "Process Manager" - would that be an option to fulfill my desired result? Or do i have to create a bundle on my own for that?

Thanks in advance for any replies!

**Comment by BlackbitDevs** [comment]
Doesn't the functionality to export the class structure already exist with the button `Export definition` when you are on the classes configuration?

Or do you want to trigger exactly this behaviour but from the main menu?

**Reply by DenisKudic** [reply]
Hello BlackbitDevs,
thank you for your reply. Actually yes, i could use the functionality of the bulk export of the class-structure. But the project will have future use-cases which still need to be defined. 

For now, i found a way to program an own function that creates the class-structure and then decode it into a desired json-format. 

Also i have created a bundle and implemented a button below the search-button in the pimcore-admin interface through the startup.js in my bundle-folder.

I struggle for 2 days now to find a way to create a "Tab" after clicking on the button. (e.g. the same way as a tab gets opened when you click on a document/asset/dataobject etc.). 

Within this tab, i only want to program a simple button which generates the json.
